### DISCUSSION TITLE
installation

### DISCUSSION THREAD

**Discussion by RuiFernandes** [directus/directus#14504] [Archive]
### Describe the Bug

just a simple install process is not working 

### To Reproduce

 npm init directus-project example-project
Need to install the following packages:
  create-directus-project
Ok to proceed? (y) y
✖ Installing Directus
Error: npm WARN config production Use `--omit=dev` instead.

### Errors Shown

 npm init directus-project example-project
Need to install the following packages:
  create-directus-project
Ok to proceed? (y) y
✖ Installing Directus
Error: npm WARN config production Use `--omit=dev` instead.
npm WARN config optional Use `--omit=optional` to exclude optional dependencies, or
npm WARN config `--include=optional` to include them.
npm WARN config 
npm WARN config     Default value does install optional deps unless otherwise omitted.
npm WARN deprecated @types/cssnano@5.1.0: This is a stub types definition. cssnano provides its own type definitions, so you do not need this installed.
npm WARN deprecated @types/pino@7.0.5: This is a stub types definition. pino provides its own type definitions, so you do not need this installed.
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated fastify-warning@0.2.0: This module renamed to process-warning
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm ERR! code 1
npm ERR! path /Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2
npm ERR! command failed
npm ERR! command sh -c node-pre-gyp install --fallback-to-build
npm ERR! CC(target) Release/obj.target/libargon2/argon2/src/ref.o
npm ERR!   CC(target) Release/obj.target/libargon2/argon2/src/argon2.o
npm ERR!   CC(target) Release/obj.target/libargon2/argon2/src/core.o
npm ERR!   CC(target) Release/obj.target/libargon2/argon2/src/blake2/blake2b.o
npm ERR!   CC(target) Release/obj.target/libargon2/argon2/src/thread.o
npm ERR!   CC(target) Release/obj.target/libargon2/argon2/src/encoding.o
npm ERR!   LIBTOOL-STATIC Release/argon2.a
npm ERR!   CXX(target) Release/obj.target/argon2/src/argon2_node.o
npm ERR! Failed to execute '/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/bin/node /Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3/argon2.node --module_name=argon2 --module_path=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3 --napi_version=8 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.9
npm ERR! node-pre-gyp info using node@16.15.1 | darwin | arm64
npm ERR! node-pre-gyp info check checked for "/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3/argon2.node" (not found)
npm ERR! node-pre-gyp http GET https://github.com/ranisalt/node-argon2/releases/download/v0.28.7/argon2-v0.28.7-napi-v3-darwin-arm64-unknown.tar.gz
npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://github.com/ranisalt/node-argon2/releases/download/v0.28.7/argon2-v0.28.7-napi-v3-darwin-arm64-unknown.tar.gz 
npm ERR! node-pre-gyp WARN Pre-built binaries not installable for argon2@0.28.7 and node@16.15.1 (node-v93 ABI, unknown) (falling back to source compile with node-gyp) 
npm ERR! node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/ranisalt/node-argon2/releases/download/v0.28.7/argon2-v0.28.7-napi-v3-darwin-arm64-unknown.tar.gz 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.0.0
npm ERR! gyp info using node@16.15.1 | darwin | arm64
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.0.0
npm ERR! gyp info using node@16.15.1 | darwin | arm64
npm ERR! gyp info find Python using Python version 3.8.9 found at "/Applications/Xcode.app/Contents/Developer/usr/bin/python3"
npm ERR! gyp info spawn /Applications/Xcode.app/Contents/Developer/usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/Users/ruifernandesguirrugo/Library/Caches/node-gyp/16.15.1/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/Users/ruifernandesguirrugo/Library/Caches/node-gyp/16.15.1',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/Users/ruifernandesguirrugo/Library/Caches/node-gyp/16.15.1/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info ok 
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.0.0
npm ERR! gyp info using node@16.15.1 | darwin | arm64
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! clang: error: no such file or directory: 'test/CMS/directus/example-project/node_modules/node-addon-api'
npm ERR! make: *** [Release/obj.target/argon2/src/argon2_node.o] Error 1
npm ERR! gyp ERR! build error 
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
npm ERR! gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Darwin 21.5.0
npm ERR! gyp ERR! command "/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/bin/node" "/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--fallback-to-build" "--module=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3/argon2.node" "--module_name=argon2" "--module_path=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3" "--napi_version=8" "--node_abi_napi=napi" "--napi_build_version=3" "--node_napi_label=napi-v3"
npm ERR! gyp ERR! cwd /Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2
npm ERR! gyp ERR! node -v v16.15.1
npm ERR! gyp ERR! node-gyp -v v9.0.0
npm ERR! gyp ERR! not ok 
npm ERR! node-pre-gyp ERR! build error 
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute '/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/bin/node /Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --fallback-to-build --module=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3/argon2.node --module_name=argon2 --module_path=/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2/lib/binding/napi-v3 --napi_version=8 --node_abi_napi=napi --napi_build_version=3 --node_napi_label=napi-v3' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/@mapbox/node-pre-gyp/lib/util/compile.js:89:23)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:527:28)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1092:16)
npm ERR! node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! node-pre-gyp ERR! System Darwin 21.5.0
npm ERR! node-pre-gyp ERR! command "/Users/ruifernandesguirrugo/.nvm/versions/node/v16.15.1/bin/node" "/Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
npm ERR! node-pre-gyp ERR! cwd /Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus/example-project/node_modules/argon2
npm ERR! node-pre-gyp ERR! node -v v16.15.1
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.9
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ruifernandesguirrugo/.npm/_logs/2022-07-19T05_38_48_709Z-debug-0.log
npm ERR! code 1
npm ERR! path /Users/ruifernandesguirrugo/Documents/Projects/Projects test/CMS/directus
npm ERR! command failed
npm ERR! command sh -c create-directus-project "example-project"

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ruifernandesguirrugo/.npm/_logs/2022-07-19T05_38_37_284Z-debug-0.log

### What version of Directus are you using?

9.0.1

### What version of Node.js are you using?

v16.15.1

### What database are you using?

n/a

### What browser are you using?

n/a

### How are you deploying Directus?

locally

**Comment by br41nslug** [comment]
If you're running an M1 you can check out this same problem https://github.com/directus/directus/discussions/5012
If not perhaps this article can help getting argon2 working https://medium.com/@su_bak/how-to-solve-argon2-install-error-when-using-npm-eef9f2613f32

**Comment by dev7ch** [comment]
Hello @br41nslug, i came here after i try to `npm install --omit=dev`. The normal installation works just fine, but i d like to create directus production docker image with minimal footprint. I am getting:

``` 

.../node_modules/.bin/directus:2
import { updateCheck } from '@directus/update-check';
^^^^^^

SyntaxError: Cannot use import statement outside a module
...
```
Is the installation of `devDependencies` required to run directus?

**Reply by br41nslug** [reply]
> Is the installation of devDependencies required to run directus?

Not as far as i know, did you check the `package.json`? Please do ask questions on discord tho instead of reviving old discussions.
